be/src/vec/functions/function_string.h

```diff
@@ -3852,4 +3852,108 @@ class FunctionIntToChar : public IFunction {
+
+        const auto* col_pos = assert_cast<const ColumnVector<Int32>*>(argument_columns[1].get())
+                                      ->get_data()
+                                      .data();
+        const auto* col_len = assert_cast<const ColumnVector<Int32>*>(argument_columns[2].get())
+                                      ->get_data()
+                                      .data();
+        const auto* col_insert = assert_cast<const ColumnString*>(argument_columns[3].get());
+
+        ColumnString::MutablePtr col_res = ColumnString::create();
+
+        if (col_const[1]) {
+            vector<true>(col_origin, col_pos, col_len, col_insert, col_res, col_const,
+                         input_rows_count);
+        } else {
+            vector<false>(col_origin, col_pos, col_len, col_insert, col_res, col_const,
+                          input_rows_count);
+        }
+
+        block.replace_by_position(result, std::move(col_res));
+        return Status::OK();
+    }
+
+private:
+    // get the new str size
+    static std::pair<bool, size_t> get_size(size_t& str_size, int& pos, int& len,
+                                            size_t& ins_size) {
+        if (pos > str_size || pos < 1) {
+            return {true, str_size};
+        }
+        if (len < 0 || pos + len - 1 >= str_size) {
+            len = str_size - pos + 1;
+            return {false, pos + ins_size - 1};
+        }
+        return {false, str_size - len + ins_size};
+    }
+
+    template <bool is_const>
+    void vector(const ColumnString* col_origin, int const* col_pos, int const* col_len,
+                const ColumnString* col_insert, ColumnString::MutablePtr& col_res,
+                bool col_const[4], size_t input_rows_count) const {
+        auto& col_res_chars = col_res->get_chars();
+        auto& col_res_offsets = col_res->get_offsets();
+        StringRef origin_str = col_origin->get_data_at(0);
+        StringRef insert_str = col_insert->get_data_at(0);
+        auto pos = col_pos[0];
+        auto len = col_len[0];
+        if constexpr (is_const) {
+            if (pos < 1 || (col_const[0] && pos > origin_str.size)) {
+                for (size_t i = 0; i < input_rows_count; i++) {
+                    origin_str = col_origin->get_data_at(index_check_const(i, col_const[0]));
+                    col_res->insert_data(origin_str.data, origin_str.size);
+                }
+                return;
+            }
+        }
+        for (size_t i = 0; i < input_rows_count; i++) {
+            origin_str = col_origin->get_data_at(index_check_const(i, col_const[0]));
+            pos = col_pos[index_check_const(i, col_const[1])];
+            len = col_len[index_check_const(i, col_const[2])];
+            insert_str = col_insert->get_data_at(index_check_const(i, col_const[3]));
+
+            if (auto [is_origin, offset] = get_size(origin_str.size, pos, len, insert_str.size);
```

Comment:
try to use branchless structure to refactor this code. now will block auto-vectorization